app/assets/javascripts/discourse/app/lib/load-emoji-search-aliases.js

```diff
@@ -0,0 +1,8 @@
+import { ajax } from "discourse/lib/ajax";
+
+let searchAliases;
+
+export default async function loadEmojiSearchAliases() {
+  searchAliases ??= await ajax("/emojis/search-aliases");
+  return searchAliases;
+}
```

Comment:
If this method is called multiple times quickly, then it'll trigger multiple ajax requests. To avoid that, I think you'd want to memoize the promise instead of the result. Something like this:
```suggestion
let searchAliasesPromise;

export default async function loadEmojiSearchAliases() {
  searchAliasesPromise ??= ajax("/emojis/search-aliases");
  return await searchAliases;
}
```

We had [a similar problem](https://github.com/discourse/discourse/commit/f0d231355ef10fd020714d661db39c7a30d8dcc5) with another memoized-async thing recently.

Reply:
Ah yes makes sense :+1: thanks